pkg/gatherers/clusterconfig/events_filtering.go

```diff
@@ -0,0 +1,76 @@
+package clusterconfig
+
+import (
+	"sort"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// filterEvents() returns events that occoured since last interval
+func getEventsForInterval(interval time.Duration, events *v1.EventList) v1.EventList {
+	oldestEventTime := time.Now().Add(-interval)
+	var filteredEvents v1.EventList
+	for i := range events.Items {
+		if isEventNew(&events.Items[i], oldestEventTime) {
+			filteredEvents.Items = append(filteredEvents.Items, events.Items[i])
+		}
+	}
+	return filteredEvents
+}
+
+// if LastTimestamp is zero then try to check the event series
+func isEventNew(event *v1.Event, oldestEventTime time.Time) bool {
+	if event.LastTimestamp.Time.After(oldestEventTime) {
+		return true
+	} else if event.LastTimestamp.IsZero() {
+		if event.Series != nil {
+			if event.Series.LastObservedTime.Time.After(oldestEventTime) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// filterAbnormalEvents returns events that have Type different from "Normal"
+func filterAbnormalEvents(events *v1.EventList) v1.EventList {
+	var filteredEvents v1.EventList
+	for i := range events.Items {
+		if isEventAbnormal(&events.Items[i]) {
+			filteredEvents.Items = append(filteredEvents.Items, events.Items[i])
+		}
+	}
+	return filteredEvents
+}
+
+func isEventAbnormal(event *v1.Event) bool {
+	return event.Type != "Normal"
+}
+
+// eventListToCompactedEventList() coverts EventList() into CompactedEventList()
```

Comment:
nit: Typo in the comment `coverts`

Reply:
Maybe better to say `converts v1.EventList into CompactedEventList` - the brackets seems not necessary when we talk about types